test/integration/PoolSwap.t.sol

```diff
@@ -1,92 +1,78 @@
+// SPDX-License-Identifier: MIT
 pragma solidity 0.8.25;
 
-import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
-import {Test} from 'forge-std/Test.sol';
-
+import {IERC20} from '@cowprotocol/interfaces/IERC20.sol';
 import {BFactory} from 'contracts/BFactory.sol';
-import {IBPool} from 'interfaces/IBPool.sol';
-
 import {GasSnapshot} from 'forge-gas-snapshot/GasSnapshot.sol';
+import {Test, Vm} from 'forge-std/Test.sol';
+import {IBFactory} from 'interfaces/IBFactory.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
 
 abstract contract PoolSwapIntegrationTest is Test, GasSnapshot {
-  BFactory public factory;
-  IBPool public pool;
+  address public pool;
+  IBFactory public factory;
 
-  IERC20 public tokenA;
-  IERC20 public tokenB;
+  IERC20 public dai = IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+  IERC20 public weth = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
 
-  address public lp = address(420);
-  address public swapper = address(69);
+  address public lp = makeAddr('lp');
 
-  function setUp() public {
-    tokenA = IERC20(address(deployMockERC20('TokenA', 'TKA', 18)));
-    tokenB = IERC20(address(deployMockERC20('TokenB', 'TKB', 18)));
+  Vm.Wallet swapper = vm.createWallet('swapper');
 
-    deal(address(tokenA), address(lp), 100e18);
-    deal(address(tokenB), address(lp), 100e18);
+  uint256 public constant HUNDRED_UNITS = 100 ether;
+  uint256 public constant ONE_UNIT = 1 ether;
+  // NOTE: hardcoded from test result
```

Comment:
If the purpose of these tests is to ensure the contracts work in a real-world scenario and we don't have any errors that are easy to miss by fuzzing (such as confusing buy and sell tokens) I'd much rather have amounts that are both simple to grasp and obvious in their correctness, and would even use ranged asserts after the swap to allow for that

when running an integration test I want to know that the returned DAI from selling 1 WETH

- is _close_ to `3.3k`, and 
- is 100% **not** `1/3.3k`

for that I'd tweak the liquidity to be similar to an existing live balancer pool with the same ratio, and be orders of magnitude greater than the swapped amount to not be bothered by slippage

If we want to for example also make sure the curve is bound more closely to the token with greater weight by doing a swap with more slippage, that should be a separate test

Reply:
To be honest I think that is too much for an integration, as you said the purpose of this tests is just test that they work in a real scenario. But lets wait for @wei3erHase opinion on this.

Reply:
for clarification, 
> If we want to for example also make sure the curve is bound more closely to the token with greater weight by doing a swap with more slippage, that should be a separate test
would 100% be out of scope for this PR

Reply:
current mkt quote is not sth we would wanna replicate (the pool is indistinct to what's current market price), but i agree we could have some more realistic numbers. 

we're deploying a 20-80 pool, which means that the pool will accept that the value should be distributed in that relation. we could make a `1000 DAI = 1 ETH` for simplicity, but then we'll have to replicate it on the pool balances to make it work.

for example, let's say the pool has 1 ETH, and that it represents the 80% of the value, on the other side we should have sth like 250 DAI, if i'm not mistaken.

```
250 * 0,8 = 200
1000 * 0,2 = 200
```

on a similar pool, but 50-50, we should expect to be pairing 1 ETH with 1000 DAI

also agree that out of scope for this PR